### PR TITLE
Fix bug 1294737: Recognize string format named placeables

### PR DESCRIPTION
**Pull request by MikkCZ**

### CODE REVIEW

**Comment by jotes** [comment]
@MikkCZ  This looks awesome -> I think we could add some tests.

e.g. Look here:
https://github.com/mozilla/pontoon/blob/master/pontoon/base/tests/test_placeables.py

Feel free to ask any questions -> I should be around today :-) Optionally you're always welcomed at #pontoon@irc.mozilla.org

**Comment by MikkCZ** [comment]
Yay! Thank you, I will add the tests tonight.

**Comment by MikkCZ** [comment]
I have added a test now and also changed the string compilation to be case insensitive.

In the tests I noticed, they are missing any case testing https://github.com/mozilla/pontoon/commit/a61a5cfbfb6d08dbf46d7c1948051bf7570bc41a. Should I add and commit under this bug?

File: pontoon/base/utils.py [hunk 1]
```diff
@@ -69,6 +69,13 @@ class SpacesPlaceable(base.Ph):
     parse = classmethod(general.regex_parse)
 
 
+class PythonFormatNamedPlaceable(base.Ph):
+    """Placeable handling named format string in python"""
+    istranslatable = False
+    regex = re.compile(r'%\([[\w\d\!\.,\[\]%:$<>\+\-= ]*\)[s|d]', re.IGNORECASE)
```
Comment (jotes) [inline pontoon/base/utils.py]: I think there's a missing %f

**Comment by jotes** [comment]
I think you can -> just add a comment.

**Comment by jotes** [comment]
You have my humble r+wc :dancer: Good job @MikkCZ !

**Comment by MikkCZ** [comment]
Good link, I will rewrite the regex accordingly. Thanks @jotes !

**Comment by MikkCZ** [comment]
@jotes r?

**Comment by jotes** [comment]
r+ :D I think we can safely pass this to @mathjazz.

**Comment by mathjazz** [comment]
Thanks for the patch, @MikkCZ! r+ :-)

Please [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) commits and we're ready to merge & deploy!

**Comment by MikkCZ** [comment]
Squashed and rebased.

**Comment by mathjazz** [comment]
Deployed.

**Pull request merged**